enhancements/cluster-api/crd-compatibility-checker.md

```diff
@@ -0,0 +1,469 @@
+#### Conditions
+
+CRDCompatibilityRequirement defines 3 conditions:
+
+* `Progressing` -
+False if the spec has been completely reconciled.
+True indicates that reconciliation is still in progress and the current status does not represent a stable state.
+Progressing False with an error reason indicates that the object cannot be reconciled.
+
+* `Admitted` -
+True if the requirement has been configured in the validating webhook, otherwise False.
+
+* `Compatible` -
+True if the observed CRD is compatible with the requirement, otherwise False.
+Note that Compatible may be False when adding a new requirement which the existing CRD does not meet.
+
+The above conditions are always set on every reconcile, and include `observedGeneration`.
+
+These conditions may have the following Reasons:
+
+* `Progressing`
+  * `ConfigurationError` -
+  This indicates that reconciliation cannot progress due to an invalid spec.
+  It is associated with a status of `False`.
+  * `TransientError` -
+  This indicates that reconciliation was prevented from completing due to an error that can be retried.
+  It is associated with a status of `True`.
+  * `UpToDate` -
+  This indicates the reconciliation completed successfully.
+  It is associated with a status of `False`.
+* `Compatible`
+  * `RequirementsNotMet` -
+  This indicates that a CRD exists, and it is not compatible with this requirement.
+  It is associated with a status of `False`.
+  * `CRDDoesNotExist` -
+  This indicates that the referenced CRD does not exist.
+  It is associated with a status of `False`.
+  * `CompatibleWithWarnings` -
+  This indicates that the CRD exists and is compatible with this requirement, but `Message` contains one or more warning messages.
+  It is associated with a status of `True`.
+  * `Compatible` -
+  This indicates that the CRD exists and is compatible with this requirement.
+  It is associated with a status of `True`.
+
+#### Validating webhooks
+
+##### ClusterResourceDefinition
+
+Create, Update, and Delete operations on CRDs will either be denied or emit a warning depending on the setting of `crdAdmitAction`.
+
+Create and Update will deny/warn if the created or updated CRD would not be compatible with any CRDCompatibilityRequirement which references it in `crdRef`.
+
+Delete will deny/warn if any CRDCompatibilityRequirement references it in `crdRef`.
+
+##### Dynamically created by `objectSchemaValidation`
+
+[!NOTE]
+Implementation of `objectSchemaValidation` will be deferred to a second phase to be implemented after the CRD validation webhook is fully implemented and integrated.
+
+If a CRDCompatibilityRequirement defines `objectSchemaValidation` we will dynamically create a new `ValidatingWebhookConfiguration` to implement it.
+This webhook will ensure that matching objects conform to the CRD given in `compatibilityCRD`, which may be different to the current CRD version.
+It will either warn or deny admission based on the setting of `objectSchemaValidation.action`.
```

Comment:
Might be helpful to provide an example of what this might mean in the real world?